extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -0,0 +1,215 @@
+			$this->title->getPrefixedText(),
+			$this->getEditorUserName() )->inLanguage( $this->getTargetLang() )->text();
+	}
+
+	public function initEmail() {
+		$this->title = \Title::newFromText( $this->request->getVal( 'title' ) );
+		$this->summary = $this->request->getVal( 'summary' );
+		$this->currentRevId = $this->request->getVal( 'currentRevId' );
+		$this->previousRevId = $this->request->getVal( 'previousRevId' );
+		$this->replyToAddress = $this->request->getVal( 'replyToAddress' );
+		$this->fromAddress = new \MailAddress( $this->request->getVal( 'fromAddress', '' ), $this->getVal( 'fromName', '' ) );
+
+		$this->assertValidParams();
+	}
+
+	/**
+	 * Validate the params passed in by the client
+	 */
+	private function assertValidParams() {
+		$this->assertValidTitle();
+		$this->assertValidRevIds();
+		$this->assertValiFromAddress();
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->title instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->title->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	private function assertValidRevIds() {
+		if ( empty( $this->currentRevId ) ) {
+			throw new Check( "Empty current Revision Id" );
+		}
+
+		if ( empty( $this->previousRevId ) ) {
+			throw new Check( "Empty previous Revision Id" );
+		}
+	}
+
+	private function assertValiFromAddress() {
+		if ( $this->fromAddress->toString() == "" ) {
+			throw new Check( "Empty from address" );
+		}
+	}
+
+	protected function getFromAddress() {
+		return $this->fromAddress;
+	}
+
+	protected function getReplyToAddress() {
+		return $this->replyToAddress;
+	}
+
+	protected function getFooterMessages() {
+		$footerMessages = [
+			wfMessage( 'emailext-watchedpage-unfollow-text',
+				$this->title->getCanonicalUrl( 'action=unwatch' ),
+				$this->title->getPrefixedText() )->inLanguage( $this->getTargetLang() )->parse()
+		];
+		return array_merge( $footerMessages, parent::getFooterMessages() );
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'articleEditedText' => $this->getArticleEditedText(),
+			'editorProfilePage' => $this->getEditorProfilePage(),
+			'editorUserName' => $this->getEditorUserName(),
+			'editorAvatarURL' => $this->getEditorAvatarUR(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getCompareChangesLabel(),
+			'buttonLink' => $this->getCompareChangesLink(),
+			'contentFooterMessages' => [
+				$this->getArticleLinkText(),
+				$this->getAllChangesText(),
+			],
+		] );
+	}
+
+	/**
+	 * @return String
+	 * Get rendered html for content unique to this email
+	 * @todo We may want to make this available more generically for other emails to use the avatar layout.
+	 */
+	protected function getContent() {
+		$css = file_get_contents( __DIR__ . '/../styles/avatarLayout.css' );
```

Comment:
As discussed, this could be a constant in this class e.g.:
```php
const LAYOUT_CSS = 'avatarLayout.css'
```
with a default constant defined in ```EmailController```.  Then  getContent could move back to the parent class and do:
```php
$css = file_get_contents( __DIR__ . '/styles/' . static::LAYOUT_CSS );
```